src/test/regress/expected/detect_conn_close.out

```diff
@@ -0,0 +1,212 @@
+ result
+---------------------------------------------------------------------
+ 4
+ 4
+(2 rows)
+
+SELECT count(*) FROM socket_test_table;
+ count
+---------------------------------------------------------------------
+   101
+(1 row)
+
+-- now, get back to sane defaults
+SET citus.max_adaptive_executor_pool_size TO 1;
+SET citus.max_cached_conns_per_worker TO 1;
+SET citus.force_max_query_parallelization  TO OFF;
+-- we can recover for modification queries as well
+-- single row INSERT
+INSERT INTO socket_test_table VALUES (1);
+SELECT result FROM run_command_on_workers($$SELECT count(*) FROM (SELECT pg_terminate_backend(pid, 1000) FROM pg_stat_activity WHERE query ilike '%socket_test_table%' AND pid !=pg_backend_pid()) as foo$$) ORDER BY result;
+ result
+---------------------------------------------------------------------
+ 1
+ 1
+(2 rows)
+
+INSERT INTO socket_test_table VALUES (1);
+-- single row UPDATE
+UPDATE socket_test_table SET value = 15 WHERE id = 1;
+SELECT result FROM run_command_on_workers($$SELECT count(*) FROM (SELECT pg_terminate_backend(pid, 1000) FROM pg_stat_activity WHERE query ilike '%socket_test_table%' AND pid !=pg_backend_pid()) as foo$$) ORDER BY result;
+ result
+---------------------------------------------------------------------
+ 0
+ 1
+(2 rows)
+
+UPDATE socket_test_table SET value = 15 WHERE id = 1;
+-- we cannot recover in a transaction block
+BEGIN;
+  SELECT count(*) FROM socket_test_table;
+ count
+---------------------------------------------------------------------
+   103
+(1 row)
+
+  -- kill all the cached backends on the workers
+  SELECT result FROM run_command_on_workers($$SELECT count(*) FROM (SELECT pg_terminate_backend(pid, 1000) FROM pg_stat_activity WHERE query ilike '%socket_test_table%' AND pid !=pg_backend_pid()) as foo$$);
+ result
+---------------------------------------------------------------------
+ 1
+ 1
+(2 rows)
+
+  SELECT count(*) FROM socket_test_table;
+ERROR:  connection to the remote node localhost:xxxxx failed with the following error:
+WARNING:  terminating connection due to administrator command
+CONTEXT:  while executing command on localhost:xxxxx
+WARNING:  terminating connection due to administrator command
+CONTEXT:  while executing command on localhost:xxxxx
+ROLLBACK;
+-- repartition joins also can recover
+SET citus.enable_repartition_joins TO on;
+SET citus.max_adaptive_executor_pool_size TO 1;
+SET citus.max_cached_conns_per_worker TO 1;
+SELECT count(*) FROM socket_test_table t1 JOIN socket_test_table t2 USING(value);
+ count
+---------------------------------------------------------------------
+   115
+(1 row)
+
+-- kill all the cached backends on the workers
+SELECT result FROM run_command_on_workers($$SELECT count(*) FROM (SELECT pg_terminate_backend(pid, 1000) FROM pg_stat_activity WHERE query ilike '%socket_test_table%' AND pid !=pg_backend_pid()) as foo$$);
+ result
+---------------------------------------------------------------------
+ 0
+ 0
```

Comment:
It's not actually killing any connections here, probably the filter is incorrect

Reply:
ah, txes changes the query to COMMIT. Now, changed the filter to gpid, now all are terminated safely

I use boolean to make sure we do not introduce any flaky tests